index.d.ts

```diff
@@ -465,25 +465,55 @@ declare namespace Moleculer {
 	};
 
 	namespace Errors {
-		class MoleculerError extends Error { }
+		class MoleculerError extends Error {
+			constructor(message: string, code: number, type: string, data: any);
+			constructor(message: string, code: number, type: string);
+			constructor(message: string, code: number);
+			constructor(message: string);
+		}
 		class MoleculerRetryableError extends MoleculerError { }
-		class MoleculerServerError extends MoleculerError { }
+		class MoleculerServerError extends MoleculerRetryableError { }
 		class MoleculerClientError extends MoleculerError { }
 
-		class ServiceNotFoundError extends MoleculerError { }
-		class ServiceNotAvailable extends MoleculerError { }
+		class ServiceNotFoundError extends MoleculerError {
+			constructor(action, nodeID);
+		}
+		class ServiceNotAvailable extends MoleculerError {
+			constructor(action, nodeID);
+		}
```

Comment:
In `ServiceNotFoundError` and `ServiceNotAvailable` the `nodeID` is optional. `action` is required. The others is correct.